### PR TITLE
Add delete pipeline item form

### PR DESCRIPTION
**Pull request by web-bert**
## Description of change

Add ability to delete an archived pipeline item. This is currently hidden behind a url.

## Test instructions

Add a company to your pipeline, edit the pipeline item and then change the url from `/edit` to `/archive`, enter a reason and archive the item. Now click edit again and change `/edit` to `/delete` then click the delete button. The item should now be removed from your pipeline.

## Screenshots

delete_form_1

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/apps/my-pipeline/client/DeletePipelineItemForm.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import React, { useEffect } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { FormActions } from 'data-hub-components'
+import { connect } from 'react-redux'
+import { RED } from 'govuk-colours'
+import WarningText from '@govuk-react/warning-text'
+import LoadingBox from '@govuk-react/loading-box'
+import Button from '@govuk-react/button'
+import Link from '@govuk-react/link'
```
Comment (ian-leggett) [inline src/apps/my-pipeline/client/DeletePipelineItemForm.jsx:10]: As Webpack will tree shake could we use `import {WarningText, LoadingBox, Button,  Link} from 'govuk-react'`

**Comment by web-bert** [comment]
> I got a 405 when trying to delete the project
> Screenshot 2020-06-22 at 12 21 27

@ian-leggett sorry, this is because we are reliant on a backend PR to get merged before this will work. I have now added labels to make this a little clearer

File: src/apps/my-pipeline/client/DeletePipelineItemForm.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          </GetPipelineData>
+        )
+      }}
+    </Task>
+  )
+}
+
+DeletePipelineItemForm.propTypes = {
+  pipelineItemId: PropTypes.string.isRequired,
+  currentPipeline: PipelineItemPropType,
```
Comment (debitan) [inline src/apps/my-pipeline/client/DeletePipelineItemForm.jsx:96]: Should this be `.required()` as well?

Comment (debitan) [reply src/apps/my-pipeline/client/DeletePipelineItemForm.jsx:96]: Or is that taken care of in the `PipelineItemPropType` file?

Comment (web-bert) [reply src/apps/my-pipeline/client/DeletePipelineItemForm.jsx:96]: Yes this is taken care of by `PipelineItemPropType`.

**Pull request merged**